influxdb3_write/src/paths.rs

```diff
@@ -0,0 +1,171 @@
+use crate::SegmentId;
+use chrono::prelude::*;
+use object_store::path::Path as ObjPath;
+use std::convert::AsRef;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// File extension for catalog files
+pub const CATALOG_FILE_EXTENSION: &str = "json";
+
+/// File extension for parquet files
+pub const PARQUET_FILE_EXTENSION: &str = "parquet";
+
+/// File extension for segment info files
+pub const SEGMENT_INFO_FILE_EXTENSION: &str = "info.json";
+
+/// File extension for segment wal files
+pub const SEGMENT_WAL_FILE_EXTENSION: &str = "wal";
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CatalogFilePath(ObjPath);
+
+impl CatalogFilePath {
+    pub fn new(segment_id: SegmentId) -> Self {
+        let path = ObjPath::from(format!(
+            "catalogs/{:010}.{}",
+            u32::MAX - segment_id.0,
+            CATALOG_FILE_EXTENSION
+        ));
+        Self(path)
+    }
+
+    pub fn dir() -> Self {
+        Self(ObjPath::from("catalogs"))
+    }
+}
+
+impl Deref for CatalogFilePath {
+    type Target = ObjPath;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl AsRef<ObjPath> for CatalogFilePath {
+    fn as_ref(&self) -> &ObjPath {
+        &self.0
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ParquetFilePath(ObjPath);
+
+impl ParquetFilePath {
+    pub fn new(db_name: &str, table_name: &str, date: DateTime<Utc>, file_number: u32) -> Self {
+        let path = ObjPath::from(format!(
+            "dbs/{db_name}/{table_name}/{}/{:010}.{}",
```

Comment:
`db_name` and `table_name` need to be converted into object store safe character set

Reply:
Implemented in [d1fbbe8](https://github.com/influxdata/influxdb/pull/24579/commits/d1fbbe8749cc4ff7aebf31d360f9e075001686a5)

Reply:
Ah, so I think what you're doing here is attempting to parse the name and returning an error if it isn't valid. That's not quite what we want. We want the `db_name` and `table_name` strings to be properly escaped so that they use valid object store names. Thus the properly escaped names would be used and the `ObjPath::parse` of the resulting name would be expected to never error.

Reply:
I think you'd use: https://docs.rs/object_store/latest/object_store/path/struct.Path.html#encode

Reply:
Ah if you look at what you linked then I could just use `from` and it should work the same.